catalog/models.py

```diff
@@ -239,6 +227,7 @@ def filter_by_products(self, products: Iterable[AbstractProduct]):
 
         return (
             self
+            .prefetch_related('tags')
```

Comment:
You are prefetching `tags` field for `Tag` model. Is it doing something valuable? It seems is not